dace/memlet.py

```diff
@@ -390,6 +391,15 @@ def try_initialize(self, sdfg: 'dace.sdfg.SDFG', state: 'dace.sdfg.SDFGState',
         if (self.data is not None and self.subset is None):
             self.subset = subsets.Range.from_array(sdfg.arrays[self.data])
 
+        other_data = None
+        if self._is_data_src and isinstance(path[-1].dst, nodes.AccessNode):
+            other_data = path[-1].dst.data
+        elif isinstance(path[0].src, nodes.AccessNode):
```

Comment:
maybe you need `not self._is_data_src` here? @alexnick83

Reply:
No, I don't think so. Here, I am trying to find the "other data," i.e., the path's destination if the memlet is a data source or the path's source if the memlet is not a data source. There is also the check that the source/destination is an access node so that we avoid trying to extract data ranges out of code nodes. Obviously, this causes a lot of issues, and it must be investigated further.

Reply:
what I mean is that the elif would still enter even if is_data_src is true here (but the destination is a code node), so it's just a boolean inversion common issue with the branch.

Reply:
🦌, thanks for pointing out